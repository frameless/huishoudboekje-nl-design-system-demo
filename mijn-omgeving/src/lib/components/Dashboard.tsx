import {Grid, GridItem} from "@chakra-ui/react";
import {Card} from "@gemeente-denhaag/card";
import {Heading2} from "@gemeente-denhaag/typography";
import React from "react";
import {useNavigate} from "react-router-dom";

const Dashboard = () => {
    const navigate = useNavigate();

    return (
        <>
            <Heading2>Huishoudboekje</Heading2>
            <Grid templateColumns={["repeat(1, 1fr)", "repeat(2, 1fr)"]} gap={10} mt={3}>
                <GridItem>
                    <Card
                        title={"Toekomst"}
                        subTitle={"Verwachte transacties"}
                        onClick={() => navigate("/toekomst")}
                    />
                </GridItem>
                <GridItem>
                    <Card
                        title={"Banktransacties"}
                        subTitle={"Overzicht van alle bij- en afschrijvingen."}
                        onClick={() => navigate("/banktransacties")}
                    />
                </GridItem>
                <GridItem>
                    <Card
                        title={"Mijn gegevens"}
                        onClick={() => navigate("/gegevens")}
                    />
                </GridItem>
                <GridItem>
                    <Card
                        title={"Afspraken"}
                        subTitle={"Overzicht van alle afspraken."}
                        onClick={() => navigate("/afspraken")}
                    />
                </GridItem>
            </Grid>
        </>
    );

};

export default Dashboard;