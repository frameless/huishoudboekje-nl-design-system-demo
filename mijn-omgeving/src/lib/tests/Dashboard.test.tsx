import {act, getAllByLabelText, getByText} from "@testing-library/react";
import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import '@testing-library/jest-dom';
import Dashboard from "../components/Dashboard";

let container: HTMLDivElement | null = null;

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockedUsedNavigate,
}));

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container!);
    container!.remove();
    container = null;
});

describe("Dashboard", () => {

    it('should check the visibility of the text', function () {
        act(() => {
            render(<Dashboard />, container);
        });

        expect(container?.innerHTML).toMatchSnapshot();

        const heading = getByText(container!, "Huishoudboekje");
        expect(heading).toBeVisible();

        const title = getByText(container!, "Toekomst");
        expect(title).toBeVisible();

        const subTitle = getByText(container!, "Verwachte transacties");
        expect(subTitle).toBeVisible();
    });

    it('should render the card component', function () {
        act(() => {
            render(<Dashboard />, container);
        });

        const cards = getAllByLabelText(container!, "ArrowRightIcon");

        cards.forEach(card => {
            expect(card).toBeVisible();

            const spy = jest.spyOn(card, "click");
            card.click();

            expect(spy).toHaveBeenCalledTimes(1);
        })
        
    });

});
