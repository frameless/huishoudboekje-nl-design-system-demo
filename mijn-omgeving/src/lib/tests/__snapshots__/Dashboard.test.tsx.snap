// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should check the visibility of the text 1`] = `"<h2 class=\\"utrecht-heading-2 utrecht-heading-2--distanced\\">Huishoudboekje</h2><div class=\\"css-1onplc2\\"><div class=\\"css-0\\"><div class=\\"MuiPaper-root MuiCard-root denhaag-card MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"denhaag-card__wrapper\\"><div class=\\"denhaag-card__background\\"></div><div class=\\"MuiCardContent-root denhaag-card__content\\"><div class=\\"denhaag-card__text-wrapper\\"><p class=\\"MuiTypography-root denhaag-card__title MuiTypography-body1\\"><a>Toekomst</a></p><p class=\\"MuiTypography-root denhaag-card__subtitle MuiTypography-body1\\">Verwachte transacties</p></div><div class=\\"MuiCardActions-root denhaag-card__actions\\"><span class=\\"material-icons MuiIcon-root denhaag-card__arrow\\" aria-hidden=\\"true\\" aria-label=\\"ArrowRightIcon\\"><svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"denhaag-icon\\" focusable=\\"false\\" aria-hidden=\\"true\\" shape-rendering=\\"auto\\"><path d=\\"M12.293 5.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L16.586 13H5a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z\\" fill=\\"currentColor\\"></path></svg></span></div></div></div></div></div><div class=\\"css-0\\"><div class=\\"MuiPaper-root MuiCard-root denhaag-card MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"denhaag-card__wrapper\\"><div class=\\"denhaag-card__background\\"></div><div class=\\"MuiCardContent-root denhaag-card__content\\"><div class=\\"denhaag-card__text-wrapper\\"><p class=\\"MuiTypography-root denhaag-card__title MuiTypography-body1\\"><a>Banktransacties</a></p><p class=\\"MuiTypography-root denhaag-card__subtitle MuiTypography-body1\\">Overzicht van alle bij- en afschrijvingen.</p></div><div class=\\"MuiCardActions-root denhaag-card__actions\\"><span class=\\"material-icons MuiIcon-root denhaag-card__arrow\\" aria-hidden=\\"true\\" aria-label=\\"ArrowRightIcon\\"><svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"denhaag-icon\\" focusable=\\"false\\" aria-hidden=\\"true\\" shape-rendering=\\"auto\\"><path d=\\"M12.293 5.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L16.586 13H5a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z\\" fill=\\"currentColor\\"></path></svg></span></div></div></div></div></div><div class=\\"css-0\\"><div class=\\"MuiPaper-root MuiCard-root denhaag-card MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"denhaag-card__wrapper\\"><div class=\\"denhaag-card__background\\"></div><div class=\\"MuiCardContent-root denhaag-card__content\\"><div class=\\"denhaag-card__text-wrapper\\"><p class=\\"MuiTypography-root denhaag-card__title MuiTypography-body1\\"><a>Mijn gegevens</a></p><p class=\\"MuiTypography-root denhaag-card__subtitle MuiTypography-body1\\"></p></div><div class=\\"MuiCardActions-root denhaag-card__actions\\"><span class=\\"material-icons MuiIcon-root denhaag-card__arrow\\" aria-hidden=\\"true\\" aria-label=\\"ArrowRightIcon\\"><svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"denhaag-icon\\" focusable=\\"false\\" aria-hidden=\\"true\\" shape-rendering=\\"auto\\"><path d=\\"M12.293 5.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L16.586 13H5a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z\\" fill=\\"currentColor\\"></path></svg></span></div></div></div></div></div><div class=\\"css-0\\"><div class=\\"MuiPaper-root MuiCard-root denhaag-card MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"denhaag-card__wrapper\\"><div class=\\"denhaag-card__background\\"></div><div class=\\"MuiCardContent-root denhaag-card__content\\"><div class=\\"denhaag-card__text-wrapper\\"><p class=\\"MuiTypography-root denhaag-card__title MuiTypography-body1\\"><a>Afspraken</a></p><p class=\\"MuiTypography-root denhaag-card__subtitle MuiTypography-body1\\">Overzicht van alle afspraken.</p></div><div class=\\"MuiCardActions-root denhaag-card__actions\\"><span class=\\"material-icons MuiIcon-root denhaag-card__arrow\\" aria-hidden=\\"true\\" aria-label=\\"ArrowRightIcon\\"><svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"denhaag-icon\\" focusable=\\"false\\" aria-hidden=\\"true\\" shape-rendering=\\"auto\\"><path d=\\"M12.293 5.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L16.586 13H5a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z\\" fill=\\"currentColor\\"></path></svg></span></div></div></div></div></div></div>"`;
